plugins/obs-filters/chroma-key-filter.c

```diff
@@ -276,6 +303,7 @@ static void chroma_key_defaults(obs_data_t *settings)
 	obs_data_set_default_int(settings, SETTING_SIMILARITY, 400);
 	obs_data_set_default_int(settings, SETTING_SMOOTHNESS, 80);
 	obs_data_set_default_int(settings, SETTING_SPILL, 100);
+	obs_data_set_default_int(settings, SETTING_SPILL_TYPE, "saturation");
```

Comment:
Setting text where an int is expected

Reply:
Oops, I completely missed that in my review. 😅

Reply:
Oof — yeah I changed this to an enum a bit last minute; will fix.

Reply:
Kind of surprising how that doesn't crash/fail to compile btw? 😅

Reply:
The default int function is accepting a long long. The compiler is probably casting the pointer address for the string to a long long implicitly. Definitely an unexpected result, but there probably should have been a compiler warning.

Reply:
Yeah, I did in fact get a compiler warning! Maybe an improvement would be to fail in CI on compiler warnings, because there don't seem to be any other warnings? In any case, thanks for catching it @exeldro!

Reply:
The project definitely has compiler warnings, both with MSVC and GCC. Most are either benign, will produce identical binaries, or come from our mixing of C and C++ and therefore aren't resolvable. Visual Studio (and I believe Xcode?) won't show warnings during a compile for parts of the project that have already been compiled and it isn't recompiling.

Reply:
Ah that makes sense! I don't know if there is something like this for C/C++, but I know that in some other languages you can run a program that will add in-line annotations for all existing warnings to suppress them. That way you can more easily turn on "fail on warnings" in CI, since it'll only happen for any future warnings (unless an explicit annotation is added to suppress them). But yeah I'm less familiar with C/C++ tooling, so not sure if something like that exists there! 😱